### PR TITLE
Pretty Disposals

### PR DESCRIPTION
**Pull request by rosieposieeee**
## About the PR

Brings back the old disposal sprites, and adds a 'space' variant that warns you that it goes to space, for the maps that have spaced disposals (Hive's perma, Sub's everything.) (Otherwise, the space one is no different than a normal one.)

If anyone has a better idea than a generic yellow warning triangle I'll sprite it.

QR7aqBY9Ru

Content Client_zmlLtlqNMP

image

## Why / Balance

The new sprites were ugly, and I had to make a new RSI for the space one anyway (or touch the upstream RSI, but I didn't want to.)

## Media

- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

**Changelog**
:cl:
- tweak: Added Space Disposals with a warning sign on them, and made disposals pretty again.

### CODE REVIEW

**Comment by DangerRevolution** [comment]
bringing back old sprite should be it's own PR for easy reversal should it be wanted/needed down the line

**Comment by DangerRevolution** [comment]
are all the pipes & unused intake sprites necessary in the new folder?

**Comment by rosieposieeee** [comment]
> bringing back old sprite should be it's own PR for easy reversal should it be wanted/needed down the line

space disposals relies on it, otherwise i would have to touch upstream files

> are all the pipes & unused intake sprites necessary in the new folder?

probably not, ill cut them out

**Comment by DangerRevolution** [comment]
> > bringing back old sprite should be it's own PR for easy reversal should it be wanted/needed down the line
> 
> space disposals relies on it, otherwise i would have to touch upstream files

you can PR the old sprites & PR the space sprites; and ask for the old sprites to be merged first so it works

**Comment by rosieposieeee** [comment]
> > > bringing back old sprite should be it's own PR for easy reversal should it be wanted/needed down the line
> > 
> > 
> > space disposals relies on it, otherwise i would have to touch upstream files
> 
> you can PR the old sprites & PR the space sprites; and ask for the old sprites to be merged first so it works

yeah but if it was undone, you'd have to undo both anyway

**Comment by IamVelcroboy** [comment]
The question is, *should* I warn the prisoners in Hive....

After all, there is an observation window set up so other prisoners can point and laugh 😄

**Comment by UnicornOnLSD** [comment]
amazing, god bless my beloved old chutes

File: Resources/Textures/DeltaV/Structures/Piping/disposal.rsi/meta.json [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+    "copyright": "https://github.com/discordia-space/CEV-Eris/blob/bbe32606902c90f5290b57d905a3f31b84dc6d7d/icons/obj/pipes/disposal.dmi and modified by DrSmugleaf, conspace & space recolors by rosieposieeee (github)",
+    "states": [
+        {
+            "name": "condisposal",
+            "directions": 1,
+            "delays": [
+                [
+                    1.0
+                ]
+            ]
```
Comment (DEATHB4DEFEAT) [inline Resources/Textures/DeltaV/Structures/Piping/disposal.rsi/meta.json]:
```suggestion
```

File: Resources/Textures/DeltaV/Structures/Piping/disposal.rsi/meta.json [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+        },
+        {
+            "name": "conmailing",
+            "directions": 1,
+            "delays": [
+                [
+                    1.0
+                ]
+            ]
```
Comment (DEATHB4DEFEAT) [inline Resources/Textures/DeltaV/Structures/Piping/disposal.rsi/meta.json]:
```suggestion
```

File: Resources/Textures/DeltaV/Structures/Piping/disposal.rsi/meta.json [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+        },
+        {
+            "name": "conspace",
+            "directions": 1,
+            "delays": [
+                [
+                    1.0
+                ]
+            ]
```
Comment (DEATHB4DEFEAT) [inline Resources/Textures/DeltaV/Structures/Piping/disposal.rsi/meta.json]:
```suggestion
```

File: Resources/Textures/DeltaV/Structures/Piping/disposal.rsi/meta.json [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+        },
+        {
+            "name": "disposal",
+            "directions": 1,
+            "delays": [
+                [
+                    1.0
+                ]
+            ]
```
Comment (DEATHB4DEFEAT) [inline Resources/Textures/DeltaV/Structures/Piping/disposal.rsi/meta.json]:
```suggestion
```

File: Resources/Textures/DeltaV/Structures/Piping/disposal.rsi/meta.json [hunk 5]
```diff
@@ -0,0 +1,164 @@
...
+        },
+        {
+            "name": "mailing",
+            "directions": 1,
+            "delays": [
+                [
+                    1.0
+                ]
+            ]
```
Comment (DEATHB4DEFEAT) [inline Resources/Textures/DeltaV/Structures/Piping/disposal.rsi/meta.json]:
```suggestion
```

File: Resources/Textures/DeltaV/Structures/Piping/disposal.rsi/meta.json [hunk 6]
```diff
@@ -0,0 +1,164 @@
...
+        },
+        {
+            "name": "space",
+            "directions": 1,
+            "delays": [
+                [
+                    1.0
+                ]
+            ]
```
Comment (DEATHB4DEFEAT) [inline Resources/Textures/DeltaV/Structures/Piping/disposal.rsi/meta.json]:
```suggestion
```

File: Resources/Textures/DeltaV/Structures/Piping/disposal.rsi/meta.json [hunk 7]
```diff
@@ -0,0 +1,164 @@
...
+        },
+        {
+            "name": "disposal-charging",
+            "directions": 1,
+            "delays": [
+                [
+                    1.0
+                ]
+            ]
```
Comment (DEATHB4DEFEAT) [inline Resources/Textures/DeltaV/Structures/Piping/disposal.rsi/meta.json]:
```suggestion
```

File: Resources/Textures/DeltaV/Structures/Piping/disposal.rsi/meta.json [hunk 8]
```diff
@@ -0,0 +1,164 @@
...
+        },
+        {
+            "name": "mailing-charging",
+            "directions": 1,
+            "delays": [
+                [
+                    1.0
+                ]
+            ]
```
Comment (DEATHB4DEFEAT) [inline Resources/Textures/DeltaV/Structures/Piping/disposal.rsi/meta.json]:
```suggestion
```

File: Resources/Textures/DeltaV/Structures/Piping/disposal.rsi/meta.json [hunk 9]
```diff
@@ -0,0 +1,164 @@
...
+                    0.1,
+                    0.1,
+                    0.1,
+                    0.1,
+                    0.5,
+                    0.1,
+                    0.1,
+                    0.1
+                ]
+            ]
```
Comment (DEATHB4DEFEAT) [inline Resources/Textures/DeltaV/Structures/Piping/disposal.rsi/meta.json]:
```suggestion
            "delays": [ [ 0.1, 0.1, 0.1, 0.1, 0.1, 0.5, 0.1, 0.1, 0.1 ] ]
```

File: Resources/Textures/DeltaV/Structures/Piping/disposal.rsi/meta.json [hunk 10]
```diff
@@ -0,0 +1,164 @@
...
```
Comment (DEATHB4DEFEAT) [inline Resources/Textures/DeltaV/Structures/Piping/disposal.rsi/meta.json]:
```suggestion
            "delays": [ [ 0.1, 0.1, 0.1, 0.1, 0.1, 0.5, 0.1, 0.1, 0.1 ] ]
```

File: Resources/Textures/DeltaV/Structures/Piping/disposal.rsi/meta.json [hunk 11]
```diff
@@ -0,0 +1,164 @@
...
+        },
+        {
+            "name": "dispover-charge",
+            "directions": 1,
+            "delays": [
+                [
+                    0.4,
+                    0.4
+                ]
+            ]
```
Comment (DEATHB4DEFEAT) [inline Resources/Textures/DeltaV/Structures/Piping/disposal.rsi/meta.json]:
```suggestion
            "delays": [ [ 0.4, 0.4 ] ]
```

File: Resources/Textures/DeltaV/Structures/Piping/disposal.rsi/meta.json [hunk 12]
```diff
@@ -0,0 +1,164 @@
...
+        },
+        {
+            "name": "dispover-full",
+            "directions": 1,
+            "delays": [
+                [
+                    0.2,
+                    0.2
+                ]
+            ]
```
Comment (DEATHB4DEFEAT) [inline Resources/Textures/DeltaV/Structures/Piping/disposal.rsi/meta.json]:
```suggestion
            "delays": [ [ 0.2, 0.2 ] ]
```

File: Resources/Textures/DeltaV/Structures/Piping/disposal.rsi/meta.json [hunk 13]
```diff
@@ -0,0 +1,164 @@
...
+        },
+        {
+            "name": "dispover-handle",
+            "directions": 1,
+            "delays": [
+                [
+                    1.0
+                ]
+            ]
```
Comment (DEATHB4DEFEAT) [inline Resources/Textures/DeltaV/Structures/Piping/disposal.rsi/meta.json]:
```suggestion
```

File: Resources/Textures/DeltaV/Structures/Piping/disposal.rsi/meta.json [hunk 14]
```diff
@@ -0,0 +1,164 @@
...
+        },
+        {
+            "name": "mailover-handle",
+            "directions": 1,
+            "delays": [
+                [
+                    1.0
+                ]
+            ]
```
Comment (DEATHB4DEFEAT) [inline Resources/Textures/DeltaV/Structures/Piping/disposal.rsi/meta.json]:
```suggestion
```

File: Resources/Textures/DeltaV/Structures/Piping/disposal.rsi/meta.json [hunk 15]
```diff
@@ -0,0 +1,164 @@
...
+        },
+        {
+            "name": "dispover-ready",
+            "directions": 1,
+            "delays": [
+                [
+                    1.0
+                ]
+            ]
```
Comment (DEATHB4DEFEAT) [inline Resources/Textures/DeltaV/Structures/Piping/disposal.rsi/meta.json]:
```suggestion
```

**Review by DEATHB4DEFEAT — CHANGES_REQUESTED** [review]

File: Resources/Textures/DeltaV/Structures/Piping/disposal.rsi/meta.json [hunk 16]
```diff
@@ -0,0 +1,75 @@
...
+    "size": {
+        "x": 32,
+        "y": 32
+    },
+    "license": "CC-BY-SA-3.0",
+    "copyright": "https://github.com/discordia-space/CEV-Eris/blob/bbe32606902c90f5290b57d905a3f31b84dc6d7d/icons/obj/pipes/disposal.dmi and modified by DrSmugleaf, conspace & space recolors by rosieposieeee (github)",
+    "states": [
+        {
+            "name": "condisposal",
+            "directions": 1,
```
Comment (FluffiestFloof) [inline Resources/Textures/DeltaV/Structures/Piping/disposal.rsi/meta.json]:
```suggestion
            "directions": 1
```

File: Resources/Textures/DeltaV/Structures/Piping/disposal.rsi/meta.json [hunk 17]
```diff
@@ -0,0 +1,75 @@
...
+        },
+        {
+            "name": "conmailing",
+            "directions": 1,
```
Comment (FluffiestFloof) [inline Resources/Textures/DeltaV/Structures/Piping/disposal.rsi/meta.json]:
```suggestion
            "directions": 1
```

File: Resources/Textures/DeltaV/Structures/Piping/disposal.rsi/meta.json [hunk 18]
```diff
@@ -0,0 +1,75 @@
...
+        },
+        {
+            "name": "conspace",
+            "directions": 1,
```
Comment (FluffiestFloof) [inline Resources/Textures/DeltaV/Structures/Piping/disposal.rsi/meta.json]:
```suggestion
            "directions": 1
```

File: Resources/Textures/DeltaV/Structures/Piping/disposal.rsi/meta.json [hunk 19]
```diff
@@ -0,0 +1,75 @@
...
+        },
+        {
+            "name": "disposal",
+            "directions": 1,
```
Comment (FluffiestFloof) [inline Resources/Textures/DeltaV/Structures/Piping/disposal.rsi/meta.json]:
```suggestion
            "directions": 1
```

File: Resources/Textures/DeltaV/Structures/Piping/disposal.rsi/meta.json [hunk 20]
```diff
@@ -0,0 +1,75 @@
...
+        },
+        {
+            "name": "mailing",
+            "directions": 1,
```
Comment (FluffiestFloof) [inline Resources/Textures/DeltaV/Structures/Piping/disposal.rsi/meta.json]:
```suggestion
            "directions": 1
```

File: Resources/Textures/DeltaV/Structures/Piping/disposal.rsi/meta.json [hunk 21]
```diff
@@ -0,0 +1,75 @@
...
+        },
+        {
+            "name": "space",
+            "directions": 1,
```
Comment (FluffiestFloof) [inline Resources/Textures/DeltaV/Structures/Piping/disposal.rsi/meta.json]:
```suggestion
            "directions": 1
```

File: Resources/Textures/DeltaV/Structures/Piping/disposal.rsi/meta.json [hunk 22]
```diff
@@ -0,0 +1,75 @@
...
+        },
+        {
+            "name": "disposal-charging",
+            "directions": 1,
```
Comment (FluffiestFloof) [inline Resources/Textures/DeltaV/Structures/Piping/disposal.rsi/meta.json]:
```suggestion
            "directions": 1
```

File: Resources/Textures/DeltaV/Structures/Piping/disposal.rsi/meta.json [hunk 23]
```diff
@@ -0,0 +1,75 @@
...
+        },
+        {
+            "name": "mailing-charging",
+            "directions": 1,
```
Comment (FluffiestFloof) [inline Resources/Textures/DeltaV/Structures/Piping/disposal.rsi/meta.json]:
```suggestion
            "directions": 1
```

File: Resources/Textures/DeltaV/Structures/Piping/disposal.rsi/meta.json [hunk 24]
```diff
@@ -0,0 +1,75 @@
...
+            "delays": [ [ 0.4, 0.4 ] ]
+        },
+        {
+            "name": "dispover-full",
+            "directions": 1,
+            "delays": [ [ 0.2, 0.2 ] ]
+        },
+        {
+            "name": "dispover-handle",
+            "directions": 1,
```
Comment (FluffiestFloof) [inline Resources/Textures/DeltaV/Structures/Piping/disposal.rsi/meta.json]:
```suggestion
            "directions": 1
```

File: Resources/Textures/DeltaV/Structures/Piping/disposal.rsi/meta.json [hunk 25]
```diff
@@ -0,0 +1,75 @@
...
+        },
+        {
+            "name": "mailover-handle",
+            "directions": 1,
```
Comment (FluffiestFloof) [inline Resources/Textures/DeltaV/Structures/Piping/disposal.rsi/meta.json]:
```suggestion
            "directions": 1
```

File: Resources/Textures/DeltaV/Structures/Piping/disposal.rsi/meta.json [hunk 26]
```diff
@@ -0,0 +1,75 @@
...
+        },
+        {
+            "name": "dispover-ready",
+            "directions": 1,
```
Comment (FluffiestFloof) [inline Resources/Textures/DeltaV/Structures/Piping/disposal.rsi/meta.json]:
```suggestion
            "directions": 1
```

**Review by FluffiestFloof — COMMENTED** [review]
should fix the tests
thanks death

**Comment by rosieposieeee** [comment]
lol

**Comment by IamVelcroboy** [comment]
Looks fine to me. Just waiting to see if any other maintainers have thoughts...?

**Review by deltanedas — DISMISSED** [review]
long live the king

File: Resources/Textures/DeltaV/Structures/Piping/disposal.rsi/meta.json [hunk 27]
```diff
@@ -0,0 +1,75 @@
...
+            "directions": 1
+        },
+        {
+            "name": "mailover-handle",
+            "directions": 1
+        },
+        {
+            "name": "dispover-ready",
+            "directions": 1
+        }
```
Comment (DEATHB4DEFEAT) [inline Resources/Textures/DeltaV/Structures/Piping/disposal.rsi/meta.json]:
```suggestion
        {
            "name": "condisposal",
        },
        {
            "name": "conmailing",
        },
        {
            "name": "conspace",
        },
        {
            "name": "disposal",
        },
        {
            "name": "mailing",
        },
        {
            "name": "space",
        },
        {
            "name": "disposal-charging",
        },
        {
            "name": "mailing-charging",
        },
        {
            "name": "disposal-flush",
            "delays": [ [ 0.1, 0.1, 0.1, 0.1, 0.1, 0.5, 0.1, 0.1, 0.1 ] ]
        },
        {
            "name": "mailing-flush",
            "delays": [ [ 0.1, 0.1, 0.1, 0.1, 0.1, 0.5, 0.1, 0.1, 0.1 ] ]
        },
        {
            "name": "dispover-charge",
            "delays": [ [ 0.4, 0.4 ] ]
        },
        {
            "name": "dispover-full",
            "delays": [ [ 0.2, 0.2 ] ]
        },
        {
            "name": "dispover-handle",
        },
        {
            "name": "mailover-handle",
        },
        {
            "name": "dispover-ready",
        }
```

**Review by DEATHB4DEFEAT — CHANGES_REQUESTED** [review]

**Review by DEATHB4DEFEAT — DISMISSED** [review]

**Comment by DEATHB4DEFEAT** [comment]
Fix tests then this looks good.

**Comment by Colin-Tel** [comment]
Any update? Tests are still failing.

File: Resources/Textures/DeltaV/Structures/Piping/disposal.rsi/meta.json [hunk 28]
```diff
@@ -0,0 +1,60 @@
...
+    "version": 1,
+    "size": {
+        "x": 32,
+        "y": 32
+    },
+    "license": "CC-BY-SA-3.0",
+    "copyright": "https://github.com/discordia-space/CEV-Eris/blob/bbe32606902c90f5290b57d905a3f31b84dc6d7d/icons/obj/pipes/disposal.dmi and modified by DrSmugleaf, conspace & space recolors by rosieposieeee (github)",
+    "states": [
+        {
+            "name": "condisposal",
```
Comment (DangerRevolution) [inline Resources/Textures/DeltaV/Structures/Piping/disposal.rsi/meta.json]:
```suggestion
            "name": "condisposal"
```

File: Resources/Textures/DeltaV/Structures/Piping/disposal.rsi/meta.json [hunk 29]
```diff
@@ -0,0 +1,60 @@
...
+        },
+        {
+            "name": "conmailing",
```
Comment (DangerRevolution) [inline Resources/Textures/DeltaV/Structures/Piping/disposal.rsi/meta.json]:
```suggestion
            "name": "conmailing"
```

File: Resources/Textures/DeltaV/Structures/Piping/disposal.rsi/meta.json [hunk 30]
```diff
@@ -0,0 +1,60 @@
...
+        },
+        {
+            "name": "conspace",
```
Comment (DangerRevolution) [inline Resources/Textures/DeltaV/Structures/Piping/disposal.rsi/meta.json]:
```suggestion
            "name": "conspace"
```

File: Resources/Textures/DeltaV/Structures/Piping/disposal.rsi/meta.json [hunk 31]
```diff
@@ -0,0 +1,60 @@
...
+        },
+        {
+            "name": "disposal",
```
Comment (DangerRevolution) [inline Resources/Textures/DeltaV/Structures/Piping/disposal.rsi/meta.json]:
```suggestion
            "name": "disposal"
```

File: Resources/Textures/DeltaV/Structures/Piping/disposal.rsi/meta.json [hunk 32]
```diff
@@ -0,0 +1,60 @@
...
+        },
+        {
+            "name": "mailing",
```
Comment (DangerRevolution) [inline Resources/Textures/DeltaV/Structures/Piping/disposal.rsi/meta.json]:
```suggestion
            "name": "mailing"
```

File: Resources/Textures/DeltaV/Structures/Piping/disposal.rsi/meta.json [hunk 33]
```diff
@@ -0,0 +1,60 @@
...
+        },
+        {
+            "name": "space",
```
Comment (DangerRevolution) [inline Resources/Textures/DeltaV/Structures/Piping/disposal.rsi/meta.json]:
```suggestion
            "name": "space"
```

File: Resources/Textures/DeltaV/Structures/Piping/disposal.rsi/meta.json [hunk 34]
```diff
@@ -0,0 +1,60 @@
...
+        },
+        {
+            "name": "disposal-charging",
```
Comment (DangerRevolution) [inline Resources/Textures/DeltaV/Structures/Piping/disposal.rsi/meta.json]:
```suggestion
            "name": "disposal-charging"
```

File: Resources/Textures/DeltaV/Structures/Piping/disposal.rsi/meta.json [hunk 35]
```diff
@@ -0,0 +1,60 @@
...
+        },
+        {
+            "name": "mailing-charging",
```
Comment (DangerRevolution) [inline Resources/Textures/DeltaV/Structures/Piping/disposal.rsi/meta.json]:
```suggestion
            "name": "mailing-charging"
```

File: Resources/Textures/DeltaV/Structures/Piping/disposal.rsi/meta.json [hunk 36]
```diff
@@ -0,0 +1,60 @@
...
+        {
+            "name": "dispover-charge",
+            "delays": [ [ 0.4, 0.4 ] ]
+        },
+        {
+            "name": "dispover-full",
+            "delays": [ [ 0.2, 0.2 ] ]
+        },
+        {
+            "name": "dispover-handle",
```
Comment (DangerRevolution) [inline Resources/Textures/DeltaV/Structures/Piping/disposal.rsi/meta.json]:
```suggestion
            "name": "dispover-handle"
```

File: Resources/Textures/DeltaV/Structures/Piping/disposal.rsi/meta.json [hunk 37]
```diff
@@ -0,0 +1,60 @@
...
+        },
+        {
+            "name": "mailover-handle",
```
Comment (DangerRevolution) [inline Resources/Textures/DeltaV/Structures/Piping/disposal.rsi/meta.json]:
```suggestion
            "name": "mailover-handle"
```

File: Resources/Textures/DeltaV/Structures/Piping/disposal.rsi/meta.json [hunk 38]
```diff
@@ -0,0 +1,60 @@
...
+        },
+        {
+            "name": "dispover-ready",
```
Comment (DangerRevolution) [inline Resources/Textures/DeltaV/Structures/Piping/disposal.rsi/meta.json]:
```suggestion
            "name": "dispover-ready"
```

**Review by DangerRevolution — COMMENTED** [review]
this should fix le PR

**Comment by rosieposieeee** [comment]
laziness ops wins again

**Comment by IamVelcroboy** [comment]
👻

**Comment by Timemaster99** [comment]
Should revive this

File: Resources/Prototypes/DeltaV/Entities/Structures/Piping/Disposal/space_disposal.yml [hunk 39]
```diff
@@ -0,0 +1,25 @@
...
+  - type: Sprite
+    sprite: DeltaV/Structures/Piping/disposal.rsi
+    snapCardinals: true
+    layers:
+    - state: condisposal
+      map: [ "enum.DisposalUnitVisualLayers.Unanchored" ]
+    - state: disposal
+      map: [ "enum.DisposalUnitVisualLayers.Base" ]
+    - state: disposal-flush
+      map: [ "enum.DisposalUnitVisualLayers.BaseFlush" ]
```
Comment (Timemaster99) [inline Resources/Prototypes/DeltaV/Entities/Structures/Piping/Disposal/space_disposal.yml]:
```suggestion
      map: [ "enum.DisposalUnitVisualLayers.OverlayFlush" ]
```
this should fix test errors

**Comment by ghost** [comment]
As the oldest open PR in the repo currently (the actual oldest one is Danger's Angle draft PR,) what's the consensus on if we want old disposals? I'm so used to the new sprite that I kinda forgot how nice this one looked.

**Comment by TadJohnson00** [comment]
I would like old disposals back, I think this is neat.

**Review by ghost — APPROVED** [review]
Yeah this looks fine. Let's try it.

**Pull request merged**